stumpy/core.py

```diff
@@ -471,19 +469,27 @@ def mass(Q, T, M_T=None, Σ_T=None):
 
     Q = np.asarray(Q)
     check_dtype(Q)
-    check_nan(Q)
 
     if Q.ndim != 1:  # pragma: no cover
         raise ValueError(f"Q is {Q.ndim}-dimensional and must be 1-dimensional. ")
+    m = Q.shape[0]
+
     T = np.asarray(T)
     check_dtype(T)
     check_nan(T)
 
     if T.ndim != 1:  # pragma: no cover
         raise ValueError(f"T is {T.ndim}-dimensional and must be 1-dimensional. ")
+    n = T.shape[0]
+
+    if np.any(np.isnan(Q)):
+        distance_profile = np.empty(n - m + 1)
+        distance_profile[
+            :
+        ] = np.inf  # Setting the Profile to inf if it cannot be calculated
+        return distance_profile
```

Comment:
This is a personal preference but I tend to like having a single return statement within a function (especially if the function is long) rather than having it short circuit and return halfway through. It could simply be an `if`/`else` statement where we store the output into `distance_profile` in both cases (i.e., set `distance_profile = calculate_distance_profile(m, QT, μ_Q.item(0), σ_Q.item(0), M_T, Σ_T)`) and return `distance_profile` only once at the end. The only thing we are really saving in your version is the intermediate storage of `distance_profile` but I think "future Sean" would thank us for being more explicit. :) 

Again, this is just a personal preference

Reply:
I changed it the way you suggested. I prefer to have short circuit return statements as you call them, because then I can directly see when the function is over for a certain branching.